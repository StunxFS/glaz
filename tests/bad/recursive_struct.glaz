struct RecursivePtr {
	a: *RecursivePtr // OK
}

struct RecursiveRef {
	a: ?&RecursiveRef // OK
}

struct Recursive {
	a: Recursive // FAIL
}

fn main() {}
