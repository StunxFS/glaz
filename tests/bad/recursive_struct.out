tests/bad/recursive_struct.glaz:10:3: error: invalid recursive struct `Recursive`
     | 
  10 | 	a: Recursive // FAIL
     | 	^
     = help: use an optional type instead: `?Recursive`
glaz error: could not compile `recursive_struct`, aborting due to previous error.
